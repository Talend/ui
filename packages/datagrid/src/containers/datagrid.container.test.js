import React from 'react';
import { shallow } from 'enzyme';

import DataGridContainer from './datagrid.container';

describe('#DataGridContainer', () => {
	it('should render DataGridContainer', () => {
		const sourceData = {};
		const wrapper = shallow(<DataGridContainer sourceData={sourceData} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
		expect(wrapper.find('DataGrid').props().data).toBe(sourceData);
		expect(DataGridContainer.displayName).toBe('Container(DataGrid)');
	});
});

describe('#DataGridContainer events', () => {
	it('should trigger onFocusedCell', () => {
		const dispatchActionCreator = jest.fn();
		const focusCellActionCreator = 'focus:cell';
		const onFocusedCell = jest.fn();
		const event = {};

		const wrapper = shallow(
			<DataGridContainer
				actionCreators={{
					onFocusedCell: focusCellActionCreator,
				}}
				dispatchActionCreator={dispatchActionCreator}
				onFocusedCell={onFocusedCell}
			/>,
		);

		wrapper
			.find('DataGrid')
			.props()
			.onFocusedCell(event);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
		expect(dispatchActionCreator).toHaveBeenCalledWith(
			focusCellActionCreator,
			{},
			{
				props: {
					actionCreators: { onFocusedCell: focusCellActionCreator },
					dispatchActionCreator,
					onFocusedCell,
				},
			},
		);
		expect(onFocusedCell).toHaveBeenCalledWith(event);
	});

	it('should not trigger onFocusedCell', () => {
		const dispatchActionCreator = jest.fn();
		const event = {};
		const wrapper = shallow(<DataGridContainer dispatchActionCreator={dispatchActionCreator} />);

		wrapper
			.find('DataGrid')
			.props()
			.onFocusedCell(event);

		// then
		expect(dispatchActionCreator).not.toHaveBeenCalled();
	});

	it('should trigger onFocusedColum', () => {
		const dispatchActionCreator = jest.fn();
		const focusColumnActionCreator = 'focus:column';
		const onFocusedColumn = jest.fn();
		const event = {};

		const wrapper = shallow(
			<DataGridContainer
				actionCreators={{
					onFocusedColumn: focusColumnActionCreator,
				}}
				dispatchActionCreator={dispatchActionCreator}
				onFocusedColumn={onFocusedColumn}
			/>,
		);

		wrapper
			.find('DataGrid')
			.props()
			.onFocusedColumn(event);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
		expect(dispatchActionCreator).toHaveBeenCalledWith(
			focusColumnActionCreator,
			{},
			{
				props: {
					actionCreators: { onFocusedColumn: focusColumnActionCreator },
					dispatchActionCreator,
					onFocusedColumn,
				},
			},
		);
		expect(onFocusedColumn).toHaveBeenCalledWith(event);
	});

	it('should not trigger onFocusedColum', () => {
		const dispatchActionCreator = jest.fn();
		const event = {};
		const wrapper = shallow(<DataGridContainer dispatchActionCreator={dispatchActionCreator} />);

		wrapper
			.find('DataGrid')
			.props()
			.onFocusedColumn(event);

		// then
		expect(dispatchActionCreator).not.toHaveBeenCalled();
	});
});
