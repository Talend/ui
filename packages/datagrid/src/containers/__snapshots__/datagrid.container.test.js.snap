// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#DataGridContainer events should trigger onFocusedCell 1`] = `
<DataGrid
  cellRenderer="DefaultCellRenderer"
  data={undefined}
  dispatchActionCreator={[Function]}
  getColumnDefsFn={[Function]}
  getPinnedColumnDefsFn={[Function]}
  getRowDataFn={[Function]}
  getValueGetterFn={[Function]}
  headerHeight={69}
  headerRenderer="DefaultHeaderRenderer"
  onFocusedCell={[Function]}
  onFocusedCellActionCreator="focus:cell"
  onFocusedColumn={[Function]}
  pinHeaderRenderer="DefaultPinHeaderRenderer"
  rowHeight={39}
  rowSelection="single"
/>
`;

exports[`#DataGridContainer events should trigger onFocusedColum 1`] = `
<DataGrid
  cellRenderer="DefaultCellRenderer"
  data={undefined}
  dispatchActionCreator={[Function]}
  getColumnDefsFn={[Function]}
  getPinnedColumnDefsFn={[Function]}
  getRowDataFn={[Function]}
  getValueGetterFn={[Function]}
  headerHeight={69}
  headerRenderer="DefaultHeaderRenderer"
  onFocusedCell={[Function]}
  onFocusedColumn={[Function]}
  onFocusedColumnActionCreator="focus:column"
  pinHeaderRenderer="DefaultPinHeaderRenderer"
  rowHeight={39}
  rowSelection="single"
/>
`;

exports[`#DataGridContainer should render DataGridContainer 1`] = `
<DataGrid
  cellRenderer="DefaultCellRenderer"
  data={Object {}}
  getColumnDefsFn={[Function]}
  getPinnedColumnDefsFn={[Function]}
  getRowDataFn={[Function]}
  getValueGetterFn={[Function]}
  headerHeight={69}
  headerRenderer="DefaultHeaderRenderer"
  onFocusedCell={[Function]}
  onFocusedColumn={[Function]}
  pinHeaderRenderer="DefaultPinHeaderRenderer"
  rowHeight={39}
  rowSelection="single"
/>
`;
